site/src/models/nfts.js

```diff
@@ -52,23 +79,59 @@ export const get = async (key) => {
 /**
  * @param {Key} key
  */
-export const remove = async (key) => stores.nfts.delete(encodeKey(key))
+export const remove = async (key) => {
+  const nft = await get(key)
+  if (!nft) return
+  await stores.nfts.delete(encodeKey(key))
+  await stores.nftsIndex.delete(
+    encodeIndexKey({ ...key, created: new Date(nft.created) })
+  )
+}
 
 /**
  * @param {any} prefix
- * @returns {Promise<NFT[]>}
+ * @param {{ limit?: number, before?: Date }} [options]
+ * @returns {Promise<Array<KVValue<NFT>>>}
  */
-export async function list(prefix) {
-  const nfts = await stores.nfts.list({
-    prefix,
-  })
-  if (nfts.keys.length > 0) {
-    return await Promise.all(
-      nfts.keys.map((key) => {
+export async function list(prefix, options) {
+  options = options || {}
+
+  const limit = Math.min(options.limit || 10, 100)
+  if (isNaN(limit) || limit < 1) {
+    throw new Error('invalid limit')
+  }
+
+  const before = options.before || new Date()
+  if (!(before instanceof Date) || isNaN(before.getTime())) {
+    throw new Error('invalid filter')
+  }
+
+  let done = false
+  let cursor
+  /** @type Promise<KVValue<NFT>>[] */
+  const nfts = []
+
+  while (!done) {
+    // @ts-ignore
+    const nftIdxList = await stores.nftsIndex.list({ prefix, cursor })
+
+    for (const k of nftIdxList.keys) {
+      const { created } = decodeIndexKey(k.name)
+      if (created.getTime() < before.getTime()) {
         // @ts-ignore
-        return stores.nfts.get(key.name).then((v) => JSON.parse(v))
-      })
-    )
+        nfts.push(stores.nfts.get(k.metadata.key, 'json'))
+        if (nfts.length >= limit) {
+          break
+        }
+      }
+    }
+
+    if (nfts.length >= limit) {
+      break
+    }
+    cursor = nftIdxList.cursor
+    done = nftIdxList.list_complete
   }
-  return []
+
+  return Promise.all(nfts)
```

Comment:
i think we need to return the cursor here, frontend needs to know the cursor to do next page right ?

Reply:
They just need to know the created date of the last item, which they can use as the `before` param for subsequent requests.